### DISCUSSION TITLE
Remove zero-to-one relationship type

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5930] [Feature Requests]
See discussion in https://github.com/specify/specify7/pull/4606#issuecomment-1981607168

The zero-to-one relationship type is a source of confusion and bugs. And no one knows anymore why it is needed (not even Ben knew why it was added in 2007)

After sp6 is deprecated, we should update the schema to change `zero-to-one` into a regular `many-to-one`

**Comment by melton-jason** [comment]
> Currently, it would be easiest for the frontend to treat zero-to-one relationships as a dependent one-to-many relationships.
> This is how the backend currently serializes and returns records with the relationship type: as an array containing one or no records.
> 
> While it would require modifications to both the frontend and backend, It might be easier to treat the relationship as a dependent many-to-one relationship, like the collectionObjectAttribute field on CollectionObject. Instead of returning an array with zero or one element, the value of the field would either be the serialized resource or null.
> 
> However, there is a small concern regarding the name of the relationships: localityDetails and geocoordDetails. The pluralized name implies there can be more than one and that the relationship is a to-many.
> So perhaps the most ideal solution is to make the relationship a true one-to-many and remove the one record maximum restriction.
> Of course, this would require compatibility with Specify 6 to be broken and assumes it makes sense to allow more than one LocalityDetail and GeocoordDetail record per Locality.

From https://github.com/specify/specify7/pull/4606#issuecomment-1983969409
